app/api/events_endpoints.rb

```diff
@@ -40,7 +40,16 @@ def represent(dataset)
         optional :with_original_date, type: Boolean, default: false
       end
       params :deleted do
-        optional :deleted, type: Boolean, default: false
+        # XXX: This is a hack for backward compatibility. It used to be Boolean
+        # and we added additional value "all".
+        #
+        #   all => return all events
+        #   true => exclude deleted events
+        #   false => exclude deleted and cancelled events (events deleted by schedule exception)
+        #
+        # This should be fixed in next API version by introducing two separate parameters:
+        # cancelled and deleted.
+        optional :deleted, type: String, regexp: /\A(true|false|yes|no|on|off|t|f|y|n|1|0|all)\z/i, default: 'false'
```

Comment:
I would prefer if we could limit the supported values only to `true`, `false` and `all`.

However, if this is needed for backward compatibility, we need specs to check whether those other strings are accepted.